commit/types.go

```diff
@@ -27,20 +27,11 @@ func DecodeCommitPluginQuery(encodedQuery []byte) (Query, error) {
 	return q, err
 }
 
-func NewCommitQuery(rmnOnRampMaxSeqNums []plugintypes.SeqNumChain, merkleRoots []cciptypes.MerkleRootChain) Query {
-	return Query{
-		RmnOnRampMaxSeqNums: rmnOnRampMaxSeqNums,
-		MerkleRoots:         merkleRoots,
-	}
-}
-
 type Observation struct {
-	MerkleRoots        []cciptypes.MerkleRootChain     `json:"merkleRoots"`
-	GasPrices          []cciptypes.GasPriceChain       `json:"gasPrices"`
-	TokenPrices        []cciptypes.TokenPrice          `json:"tokenPrices"`
-	OnRampMaxSeqNums   []plugintypes.SeqNumChain       `json:"onRampMaxSeqNums"`
-	OffRampNextSeqNums []plugintypes.SeqNumChain       `json:"offRampNextSeqNums"`
-	FChain             map[cciptypes.ChainSelector]int `json:"fChain"`
+	MerkleObs merkleroot.Observation          `json:"merkleObs"`
+	TokenObs  tokenprice.Observation          `json:"tokenObs"`
+	GasObs    chainfee.Observation            `json:"gasObs"`
+	FChain    map[cciptypes.ChainSelector]int `json:"fChain"`
```

Comment:
The next logical step is to have this top level plugin fully generic, I suppose this would end up as `map[string]any`.

Reply:
I like the idea